tools/c7n_oci/c7n_oci/resources/compute.py

```diff
@@ -0,0 +1,296 @@
+in data corruption. To avoid this, manually shut down the instance using the commands available in the OS before you softstop the
+instance.
+
+- **SOFTRESET** - Gracefully reboots the instance by sending a shutdown command to the operating system.
+After waiting 15 minutes for the OS to shut down, the instance is powered off and
+then powered back on.
+
+
+- **SENDDIAGNOSTICINTERRUPT** - For advanced users. **Caution: Sending a diagnostic interrupt to a live system can
+cause data corruption or system failure.** Sends a diagnostic interrupt that causes the instance's
+OS to crash and then reboot. Before you send a diagnostic interrupt, you must configure the instance to generate a
+crash dump file when it crashes. The crash dump captures information about the state of the OS at the time of
+the crash. After the OS restarts, you can analyze the crash dump to diagnose the issue. For more information, see
+[Sending a Diagnostic Interrupt](/iaas/Content/Compute/Tasks/sendingdiagnosticinterrupt.htm).
+
+
+
+- **DIAGNOSTICREBOOT** - Powers off the instance, rebuilds it, and then powers it back on.
+Before you send a diagnostic reboot, restart the instance's OS, confirm that the instance and networking settings are configured
+correctly, and try other [troubleshooting steps](/iaas/Content/Compute/References/troubleshooting-compute-instances.htm).
+Use diagnostic reboot as a final attempt to troubleshoot an unreachable instance. For virtual machine (VM) instances only.
+For more information, see [Performing a Diagnostic Reboot](/iaas/Content/Compute/Tasks/diagnostic-reboot.htm).
+
+
+- **REBOOTMIGRATE** - Powers off the instance, moves it to new hardware, and then powers it back on. For more information, see
+[Infrastructure Maintenance](/iaas/Content/Compute/References/infrastructure-maintenance.htm).
+
+
+For more information about managing instance lifecycle states, see
+[Stopping and Starting an Instance](/iaas/Content/Compute/Tasks/restartinginstance.htm).
+
+
+    Please refer to the Oracle Cloud Infrastructure Python SDK documentation for parameter details to this action
+    https://docs.oracle.com/en-us/iaas/tools/python/latest/api/core/client/oci.core.ComputeClient.html#oci.core.ComputeClient.instance_action
+
+    .. code-block:: yaml
+
+        policies:
+            - name: perform-instance-action-action
+              resource: oci.instance
+              actions:
+                - type: instance_action
+
+    """
+    schema = type_schema('instance_action',
+                         params={'type': 'object'},
+                         rinherit=OCIBaseAction.schema)
+
+    def perform_action(self, resource):
+        client = self.manager.get_client()
+        params_dict = {}
+        params_model = {}
+        additional_details = resource.get('additional_details')
+        if self.data.get('params') and self.data.get('params').get('instance_id'):
```

Comment:
we wouldn't ever typically pass in an instance id to a policy action, a user could do that if they wanted via a filter, but actions are typically intended to operate on the filtered resources they are passed in.